app/src/sharedTestUtils/java/ch/epfl/favo/view/MockDatabaseWrapper.java

```diff
@@ -60,4 +68,19 @@ public void removeDocument(String key) {}
     else future.complete(mockDocument);
     return future;
   }
+
+  @Override
+  public CompletableFuture<List<T>> getAllDocumentsLongitudeLatitudeBounded(Location loc, double radius, Activity activity) {
+    CompletableFuture<List<T>> future = new CompletableFuture<>();
+    if (throwError) future.completeExceptionally(new RuntimeException());
+    else {
+      ArrayList<T> arrayList = new ArrayList<>();

```

Comment:
FakeItemFactory also produces a list of fake favors in case you want to use that